src/plugins/discover/public/application/apps/main/services/use_saved_search.ts

```diff
@@ -164,6 +164,13 @@ export const useSavedSearch = ({
       searchSessionManager.newSearchSessionIdFromURL$.pipe(filter((sessionId) => !!sessionId))
     ).pipe(debounceTime(100));
 
+    /**
+     * Skip initial fetch when discover:searchOnPageLoad is disabled.
+     */
+    if (initialFetchStatus === FetchStatus.UNINITIALIZED) {
+      fetch$ = fetch$.pipe(startWith()).pipe(skip(1));
```

Comment:
```suggestion
      fetch$ = fetch$.pipe(startWith(), skip(1));
```

Reply:
Also: can you elaborate on why we need the `startWith` here? Isn't it enough just to `skip(1)`?

Reply:
Well, skipping first emit works correctly for dirty state, when timefilter options or query was not changed, since we will receive an emit from thouse things initially. Then `debounce` will squash all events into one, so by `pipe(skip(1))` we will skip initial fetch.

On the other hand, when we have default state (nothig changed), we will not get initial emit from any of the observables. That is why we emitting manually with `startWith(undefined)` and skip it straight away.